### ISSUE TITLE
We should verify that the *primary* email because the MDN "main" email

### ISSUE THREAD

**Issue by peterbe** [mdn/kuma#6541]
If you use GitHub as a provider, as part of the OAuth2 flow, they send us a LIST of emails it looks something like this:

```json
{
   "email_addresses": [
      {
         "email": "escattune@gmail.com",
         "primary": true,
         "verified": true,
         "visibility": "public"
      },
      {
         "email": "rjohnsun@mozilla.com",
         "primary": false,
         "verified": true,
         "visibility": null
      }
   ]
}
```

In MDN, with `django-allauth` we store this in two places:

* User instance
* EmailAddress(es)

The `user.email` is what we call the "MDN main email" and it's always expected to be one of `[x.email for x in user.emailaddress_set.all()]`.

This issue is about checking that the right email address gets put into the main email on the user. Not just the first in the list. Because the first one might be `"primary": false`. 

This problem is unique to GitHub, as a provider, we think since it appears that with Google you always only get a simple `"email": EMAIL`.

**Comment by peterbe** [comment]
Note! It's unlikely that to be safe to start on this right away. A lot of discussions and decisions are to be made about how email and signup works and there might be significant code overlap that could cause confusion. See https://docs.google.com/document/d/17FUhSfPQ5uPyA5VvctYFZZ0tPhc205gr4nZM8Nsu2KI/edit# (if you have access)

**Comment by peterbe** [comment]
When working on https://github.com/mdn/kuma/issues/6548 I think we're going to get this right automatically. 
We already have a jinja helper function that looks like this:

```python
@library.global_function
def get_primary_email(email_addresses):
    for email in email_addresses:
        if email_addresses[email].get("primary"):
            return email
```

It gets used in the `signup.html` like this:

```
{% set primary_email = get_primary_email(email_addresses) %}
<input type="hidden" name="email" value="{{ primary_email or default_email }}">
```
which means that the *first* email that is *primary* becomes the `user.email` eventually. 

We *might* do some backflips to try to write a unit test that checks what the `user.email` becomes in relation to `user.emailaddress_set.all()` but I really don't think it's worth it if you ask me.

**Comment by peterbe** [comment]
Ah! I forgot. When I worked on https://github.com/mdn/kuma/pull/6561 I *did* verify that *this* issue would be solved. In particular, I made one of the tests deliberately mess with the order of the email address from GitHub OAuth. And it worked! 
I just forgot to set the `Fixes #6541` on the PR.
